### DISCUSSION TITLE
Repeated verifications not supported?

### DISCUSSION THREAD

**Discussion by flandreas** [lupuuss/Mokkery#6] [Q&A] [answered]
First, thank you for this library! I'm on my way to migrate my application from mockk to mockery.

I'm currently struggling to understand how `verify`works. To me, it looks like `verify` cannot be repeatedly called.

```
@Test
fun test() {
    val list = mock<MutableList<String>>(autofill)

    list.add("1")

    verify(exactly(1)) { list.add(any()) }
    verify(exactly(1)) { list.add(any()) }
}
```

The second call of `verify` fails with
```
Expected exactly 1 calls, but no matching calls for MutableList(1).add(element = any())!
```

In mockk, the equivalent test succeeds:
```
@Test
fun test() {
    val list = mockk<MutableList<String>>(relaxed = true)

    list.add("1")

    verify(exactly = 1) { list.add(any()) }
    verify(exactly = 1) { list.add(any()) }
}
```

It's probably just me who doesn't get it, but does `verify {}` reset a mock, something like `resetCalls()`does?

**Comment by lupuuss** [comment, marked as answer]
Hi! `verify` does not reset the mock. Instead, it operates on *unverified* calls, so previously verified calls are ignored. I should probably add it to the documentation as it's not mentioned anywhere. Do you find this behavior problematic?

**Reply by flandreas** [reply]
Thanks. I don't find the behaviour problematic, you just have to be aware of it. Adding it to the documentation would surely be useful.

**Reply by lupuuss** [reply]
[Documentation updated!](https://mokkery.dev/docs/Guides/Verifying/) Thank you for raising the issue!
